plugins/authentication/request-error.js

```diff
@@ -12,7 +12,11 @@ function authenticationRequestError (state, error, options) {
   }
 
   if (error.status === 401 && otpRequired && error.request && error.request.headers['x-github-otp']) {
-    throw new HttpError('Invalid one-time password for two-factor authentication', 401, error.headers, options)
+    if (state.otp) {
+      delete state.otp // no longer valid, request again
+    } else {
+      throw new HttpError('Invalid one-time password for two-factor authentication', 401, error.headers, options)
+    }
```

Comment:
### 🤔 

This is more tricky than I thought. Right now we set `state.otp` right after `on2fa()` was resolved, we don’t know at this point if the OTP is valid or not. If it’s invalid, then the "no longer valid" is incorrect, it was never valid before. This is why the "basic with 2fa and invalid one-time-password" test is currently failing, a 3rd request is being sent because the code above no longer throws an error, instead it deletes `state.otp` and then sets it again from the `on2fa` method.

I _think_ what the correct thing to do would be to set `state.otp = oneTimePassword` after `state.octokit.request(options)` resolved

```diff
     .then((oneTimePassword) => {
-      state.otp = oneTimePassword
       return state.octokit.request(options)
+        .then(response => {
+          state.otp = oneTimePassword
+          return response
+        })
     })
```

Now that `state.otp` is set after `state.octokit.request(options)` we would need to set the OTP header for this one request manually. 

The benefit would be that the request would fail right away if the OTP is incorrect. If `state.otp` is set we can be sure that it was valid before and run the `auth.on2fa` function again.

Does that make sense?